### PR TITLE
[Flink AI Flow] [Refractor] Unify factory plugin signature of scheduler and blob manager

### PR DESCRIPTION
**Pull request by bgeng777**
## What is the purpose of the change

Unify factory plugin signature of scheduler and blob manager

## Brief change log

1. Unify factory plugin signature of scheduler and blob manager into following format:
 ```
def create_xxxr(cls, class_name, config: Dict) -> xxx:
.....
```
2. Rename `scheduler_class_name` to `scheduler_class` in SchedulerServiceConfig

## Verifying this change
This change is already covered by existing tests, such as *(please describe tests)*.

### CODE REVIEW

File: flink-ai-flow/ai_flow/plugin_interface/blob_manager_interface.py [hunk 1]
```diff
@@ -18,9 +18,40 @@
 #
 from abc import ABC, abstractmethod
 from typing import Text, Dict
+
+from ai_flow.common.configuration import AIFlowConfiguration
 from ai_flow.common.module_load import import_string
 
 
+class BlobConfig(AIFlowConfiguration):
```
Comment (jiangxin369) [inline flink-ai-flow/ai_flow/plugin_interface/blob_manager_interface.py:26]: Now that the `BlobConfig` and `BlobManagerFactory` are defined in `blob_manager_interface.py`, we'd better move `SchedulerFactory` and `SchedulerServiceConfig` to `scheduler_interface.py`

File: flink-ai-flow/ai_flow/plugin_interface/blob_manager_interface.py [hunk 2]
```diff
@@ -18,9 +18,40 @@
...
+
+    def __init__(self, config: Dict):
+        super().__init__()
+        self['blob_manager_class'] = None
+        self['blob_manager_config'] = {}
+        if config is None or 'blob_manager_class' not in config:
```
Comment (jiangxin369) [inline flink-ai-flow/ai_flow/plugin_interface/blob_manager_interface.py:32]: Why `BlobConfig` inits with a Dict and checks correctness inner init(), while `SchedulerServiceConfig` inits with nothing and the caller needs to check correctness themselves ?

File: flink-ai-flow/ai_flow_plugins/tests/blob_manager_plugins/test_oss_blob_manager.py [hunk 3]
```diff
@@ -33,15 +33,19 @@ def test_project_upload_download_oss(self):
...
             'blob_manager_class': 'ai_flow_plugins.blob_manager_plugins.oss_blob_manager.OssBlobManager',
-            'local_repository': '/tmp',
-            'access_key_id': os.environ.get('access_key_id'),
-            'access_key_secret': os.environ.get('access_key_secret'),
-            'endpoint': os.environ.get('endpoint'),
-            'bucket': os.environ.get('bucket'),
-            'repo_name': os.environ.get('repo_name')
+            'blob_manager_config': {
+                'local_repository': '/tmp',
+                'access_key_id': os.environ.get('access_key_id'),
```
Comment (jiangxin369) [inline flink-ai-flow/ai_flow_plugins/tests/blob_manager_plugins/test_oss_blob_manager.py:38]:
```suggestion
                'access_key_id': os.environ.get('blob_server.access_key_id'),
```

File: .gitignore [hunk 4]
```diff
@@ -57,6 +57,7 @@ flink-ai-flow/ai_flow.egg-info/*
 flink-ai-flow/ai_flow/test/endpoint/aiflow.db
 flink-ai-flow/ai_flow/test/api/workflow_config.json
 flink-ai-flow/ai_flow/test/common/test.yaml
+flink-ai-flow/examples/*/generated/*
 flink-ai-flow/examples/simple_examples/logs/*
 flink-ai-flow/examples/simple_examples/temp/*
```
[the remark this answers is not in the document]
Comment (bgeng777) [reply .gitignore:62]: Removed

File: flink-ai-flow/ai_flow/plugin_interface/blob_manager_interface.py (hunk 1 above)
Comment (bgeng777) [reply flink-ai-flow/ai_flow/plugin_interface/blob_manager_interface.py:26]: I create an issue for this. And I think it is better to start this refractor in a new pr due to such changes is relevant to a lot of files and needs to be more careful.

File: flink-ai-flow/ai_flow_plugins/tests/blob_manager_plugins/test_oss_blob_manager.py (hunk 3 above)
Comment (bgeng777) [reply flink-ai-flow/ai_flow_plugins/tests/blob_manager_plugins/test_oss_blob_manager.py:38]: Fixed

File: flink-ai-flow/ai_flow/plugin_interface/blob_manager_interface.py (hunk 2 above)
Comment (bgeng777) [reply flink-ai-flow/ai_flow/plugin_interface/blob_manager_interface.py:32]: I agree it is better to do check in the __init__ class but I think it is better to do such change in next pr which includes the refractor of scheduler plugin.

**Review by jiangxin369 — APPROVED** [review]

**Pull request merged**